### PR TITLE
Add Temperature sensor example

### PR DESCRIPTION
**Pull request by ChuefengVang**
Ignore this for now until implementation is pushed.

### CODE REVIEW

**Comment by ChuefengVang** [comment]
Should be good now.

File: firmware/examples/Temperature/source/main.cpp [hunk 1]
```diff
@@ -1,20 +1,22 @@
-#include <inttypes.h>
 #include "L2_HAL/sensors/temperature/temperature.hpp"
 #include "utility/log.hpp"
 
 int main(void)
 {
   Temperature temperature_example;
   bool turn_on = temperature_example.Initialize();
-  DEBUG_PRINT("Temperature Device set: %i \n", turn_on);
+  DEBUG_PRINT("Temperature Device set: %d \n", turn_on);
```
Comment (kammce) [inline firmware/examples/Temperature/source/main.cpp]: use `LOG_INFO`, do not use DEBUG_PRINT.

File: firmware/examples/Temperature/source/main.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  {
+    LOG_INFO("Temperature Device is ON.\n");
+    double data_cast = static_cast<double>(temperature_example.GetCelsius());
+    LOG_INFO("Celsius Data: %0.2g \n", data_cast);
+
+    // This will only return Fahrenheit.
+    data_cast = static_cast<double>(temperature_example.GetFahrenheit());
+    LOG_INFO("Fahrenheit: %0.2g \n\n", data_cast);
+  }
+
```
Comment (kammce) [inline firmware/examples/Temperature/source/main.cpp]: Remove this line.

File: firmware/examples/Temperature/source/main.cpp [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#include "L2_HAL/sensors/temperature/temperature.hpp"
+#include "utility/log.hpp"
+
+int main(void)
+{
+  Temperature temperature_example;
+  bool temperature_sensor_initialized = temperature_example.Initialize();
+  LOG_INFO("Temperature Device set: %d \n", temperature_sensor_initialized);
+
+  if (temperature_sensor_initialized)
```
Comment (kammce) [inline firmware/examples/Temperature/source/main.cpp]: Can you make this a loop so users can see the temperature data changing in realtime?

File: firmware/examples/Temperature/source/main.cpp [hunk 4]
```diff
@@ -0,0 +1,30 @@
+#include "L2_HAL/sensors/temperature/temperature.hpp"
+#include "utility/log.hpp"
+
+int main(void)
+{
+  Temperature temperature_example;
+  bool temperature_sensor_initialized = temperature_example.Initialize();
+  LOG_INFO("Temperature Device set: %d \n", temperature_sensor_initialized);
```
Comment (kammce) [inline firmware/examples/Temperature/source/main.cpp]: Remove this line. Not necessary, and is covered by the if-else below.

**Pull request merged**